Sources/SwiftParser/Directives.swift

```diff
@@ -163,10 +163,25 @@ extension Parser {
     }
 
     let (unexpectedBeforePoundEndIf, poundEndIf) = self.expect(.poundEndifKeyword)
+    var unexpectedAfterPoundEndif: RawUnexpectedNodesSyntax?
+    if !self.currentToken.isAtStartOfLine {
+      var unexpectedTokens = [RawTokenSyntax]()
+      var loopProgress = LoopProgressCondition()
+      while !self.at(.eof), loopProgress.evaluate(self.currentToken) {
+        if !currentToken.isAtStartOfLine {
+          unexpectedTokens += [self.consumeAnyToken()]
+        } else {
+          break
+        }
+      }
```

Comment:
Can’t you simplify this?

```suggestion
      while !self.at(.eof), !currentToken.isAtStartOfLine, loopProgress.evaluate(self.currentToken) {
        unexpectedTokens.append(self.consumeAnyToken())
      }
```